codyze-core/src/main/kotlin/de/fraunhofer/aisec/codyze/Project.kt

```diff
@@ -171,17 +179,33 @@ class AnalysisProject(
                                 }
                                 .toMutableMap()
                         )
-                        .topLevel(componentDir)
+                        .topLevels(it.associate { Pair(it, componentDir.resolve(it)) })
             }
 
-            translationOptions.exclusionPatterns?.forEach {
-                builder = builder.exclusionPatterns(it)
-            }
+            exclusionPatterns?.forEach { builder = builder.exclusionPatterns(it) }
+            configBuilder?.invoke(builder)
 
             return AnalysisProject(
                 config = builder.build(),
-                name = projectOptions.directory.fileName.toString(),
-                projectDir = projectOptions.directory,
+                name = projectDir.fileName.toString(),
+                projectDir = projectDir,
+            )
+        }
+
+        /** Builds a translation configuration from the given CLI options. */
+        fun fromOptions(
+            projectOptions: ProjectOptions,
+            translationOptions: TranslationOptions,
+            configBuilder:
```

Comment:
I find the name of `configModifier` from [TestUtils.kt#L132](https://github.com/Fraunhofer-AISEC/cpg/blob/48bc1a0bf4a5abf9f0f06f81abff742222d9ab30/cpg-core/src/testFixtures/kotlin/de/fraunhofer/aisec/cpg/test/TestUtils.kt#L132) more suitable. We may need it to be more specific and call it cpgConfigModifier to differentiate it from the Codyze configuration.

Any thoughts?